app/views/unattended/provisioning_templates/snippet/puppet.conf.erb

```diff
@@ -41,9 +41,9 @@ ssldir = <%= ssl_dir %>
 [agent]
 pluginsync      = true
 report          = true
-<%- if @host.puppet_ca_server.strip -%>
-ca_server       = <%= @host.puppet_ca_server %>
+<%- if host_puppet_ca_server.present? -%>
+ca_server       = <%= host_puppet_ca_server %>
 <%- end -%>
 certname        = <%= @host.certname %>
-environment     = <%= @host.environment %>
-server          = <%= @host.puppetmaster %>
+environment     = <%= host_puppet_environment %>
+server          = <%= host_puppet_server %>
```

Comment:
And this as well. Note that if no server is configured, Puppet has other discovery mechanisms that users might use (DNS through SRV records or just "puppet" for example.